src/ivectorbin/compute-vad-from-frame-likes.cc

```diff
@@ -0,0 +1,212 @@
+       0 0
+       1 1
+       2 0
+*/
+void PrepareMap(const std::string &map_rxfilename, int32 num_classes,
+                unordered_map<int32, int32> *map) {
+  Input map_input(map_rxfilename);
+  for (int32 i = 0; i < num_classes; i++)
+    (*map)[i] = i;
+
+  if (!map_rxfilename.empty()) {
+    std::string line;
+    while (std::getline(map_input.Stream(), line)) {
+      if (line.size() == 0) continue;
+      int32 start = line.find_first_not_of(" \t");
+      int32 end = line.find_first_of('#'); // Ignore trailing comments
+      if (start == std::string::npos || start == end) continue;
+      end = line.find_last_not_of(" \t", end - 1);
+      KALDI_ASSERT(end >= start);
+      std::vector<std::string> fields;
+      SplitStringToVector(line.substr(start, end - start + 1),
+         " \t\n\r", true, &fields);
+      if (fields.size() != 2) {
+        KALDI_ERR << "Bad line. Expected two fields, got: "
+                  << line;
+      }
+      (*map)[std::atoi(fields[0].c_str())] = std::atoi(fields[1].c_str());
+    }
+  }
+
+  if (map->size() > num_classes)
+    KALDI_ERR << "Map table has " << map->size() << " classes.  "
+              << "Expected " << num_classes << " or fewer";
+}
+
+/**
+   PreparePriors creates a table specifying the priors for each class.
+   If priors_str is empty, uniform priors are assumed.  If priors_str is
+   nonempty, the comma-separated floats are parsed out.  If present, the
+   input of priors_str is of the form:
+      0.5,0.25,0.25
+*/
+void PreparePriors(const std::string &priors_str, int32 num_classes,
+                std::vector<BaseFloat> *priors) {
+  if (priors_str.empty()) {
+    for (int32 i = 0; i < num_classes; i++)
+      priors->push_back(log(1.0/num_classes)); // Uniform priors
+  } else {
+    SplitStringToFloats(priors_str, ",", false, priors);
+    for (int32 i = 0; i < priors->size(); i++)
+      (*priors)[i] = log((*priors)[i]);
+  }
+
+  if (priors->size() != num_classes)
+    KALDI_ERR << "Prior table has " << priors->size() << " classes.  "
+              << "Expected " << num_classes;
+}
+
+}
+
+int main(int argc, char *argv[]) {
+  using namespace kaldi;
+  typedef kaldi::int32 int32;
+  try {
+    const char *usage =
+      "This program computes frame-level voice activity decisions from a\n"
+      "set of input frame-level log-likelihoods.  Frames are assigned\n"
+      "labels according to the class for which the log-likelihood\n"
+      "(optionally weighted by a prior) is maximal.  The class labels are\n"
+      "determined by the order of inputs on the command line.  See options\n"
+      "for more details.\n"
+      "See also: fgmm-global-get-frame-likes, compute-vad, merge-vads\n"
+      "\n"
+      "Usage: compute-vad-from-frame-likes [options] <likes-rspecifier-1>\n"
+      "    ... <likes-rspecifier-n> <vad-wspecifier>\n"
+      "e.g.: compute-vad-from-frame-likes --map=label_map.txt\n"
+      "    scp:likes1.scp scp:likes2.scp ark:vad.ark\n";
```

Comment:
It might be a good idea to mention explicitly that fgmm-global-get-frame-likes might be where you'd get the frame likelihoods from.